### PR TITLE
CA-387456 serialise Pool.eject

### PR DESCRIPTION
**Pull request by lindig**
When a host is ejected from a pool, its certificate for pool-internal communication is removed from all remaining hosts. This is unlikely to work when we have multiple eject operations overlapping. Hence, introduce an eject operation and make sure ejects are serialised.

### CODE REVIEW

File: ocaml/idl/datamodel_lifecycle.ml [hunk 1]
```diff
@@ -30,7 +30,7 @@ let prototyped_of_field = function
   | "VTPM", "persistence_backend" ->
       Some "22.26.0"
   | "host", "numa_affinity_policy" ->
-      Some "23.32.0-next"
+      Some "24.2.0"
```
Comment (robhoes) [inline ocaml/idl/datamodel_lifecycle.ml:33]: Ah, could you change these here and below to 24.0.0 please?

Comment (lindig) [reply ocaml/idl/datamodel_lifecycle.ml:33]: How about removing these changes entirely (I usually don't include them)?

**Review by robhoes — APPROVED** [review]
This looks right. Have you verified that parallel ejects behave correctly without raising errors? You should see OTHER_OPERATION_IN_PROGRESS in the coordinator logs while an eject call is being "queued".

File: ocaml/idl/datamodel_lifecycle.ml (hunk 1 above)
Comment (robhoes) [reply ocaml/idl/datamodel_lifecycle.ml:33]: It needs to be done eventually...

**Comment by lindig** [comment]
Yes, I see them:
```
api: [debug||5737 /var/lib/xcp/xapi|pool.eject R:54b37a4a4896|helpers] Pool.eject locking failed: caught transient failure OTHER_OPERATION_IN_PROGRESS: [ pool; OpaqueRef:52251de1-4366-6c23-2ca3-c16ae442f0c2 ]
api: [debug||5737 /var/lib/xcp/xapi|pool.eject R:54b37a4a4896|helpers] Waiting for up to 5.998897 seconds before retrying...
```

File: ocaml/idl/schematest.ml [hunk 2]
```diff
@@ -3,7 +3,7 @@ let hash x = Digest.string x |> Digest.to_hex
 (* BEWARE: if this changes, check that schema has been bumped accordingly in
    ocaml/idl/datamodel_common.ml, usually schema_minor_vsn *)
```
Comment (psafont) [inline ocaml/idl/schematest.ml:4]: please change schema_minor_vsn to 772

**Review by psafont — APPROVED** [review]

**Comment by lindig** [comment]
@minglumlu I have not looked; this one was found in testing. I assume joining is less critical with respect to certificates because we are adding capabilities rather taken them away.

**Comment by psafont** [comment]
I did take into account parallel join when designing the feature, and we took care in ensuring it works in any case.

In particular, xapi checks for missing certificates on startup and downloads them from the coordinator

**Pull request merged**